### ISSUE TITLE
RUBYGEMS_GEMDEPS breaks globally installed gems not in Gemfile

### ISSUE THREAD

**Issue by martinisoft** [rubygems/rubygems#1034]
I've uncovered a issue when setting RUBYGEMS_GEMDEPS with '-' as it creates a weird loading error with any globally installed gem binaries such as homebrew, vagrant, and chef. This issue is also present in the recently release 2.4.2 rubygems version as well.

Example Gemfile from [chef-rvm](https://github.com/fnichol/chef-rvm):
```ruby
source 'https://rubygems.org'

gem 'foodcritic', '~> 3.0.3'
gem 'minitest'
gem 'rake'
gem 'rspec', '= 3.0.0'
gem 'guard-rspec'

# allow CI to override the version of Chef for matrix testing
gem 'chef', (ENV['CHEF_VERSION'] || '>= 0.10.10')

group :development do
  gem 'emeril'
end

group :integration do
  gem 'berkshelf', '~> 3.1.5'
  gem 'test-kitchen', '~> 1.2.1'
  gem 'kitchen-vagrant', '~> 0.15.0'
end
```

An example backtrace when running the brew command in the directory containing this Gemfile:
```
% brew                                                                                     {1}
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/dependency_resolver.rb:490:in `resolve_for': unable to find any gem matching dependency 'foodcritic (~> 3.0.3)' (Gem::UnsatisfiableDepedencyError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/dependency_resolver.rb:286:in `resolve'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/request_set.rb:39:in `resolve'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/request_set.rb:47:in `resolve_current'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems.rb:244:in `detect_gemdeps'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems.rb:1109:in `<top (required)>'
	from <internal:gem_prelude>:1:in `require'
	from <internal:gem_prelude>:1:in `<compiled>'
```

**Comment by acook** [comment]
I wanted to add that it causes more issues than just with `brew` and the like, it also causes issues with `gem` itself. 

~~~
$ brew doctor
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/dependency_resolver.rb:490:in `resolve_for': unable to find any gem matching dependency 'rails (= 3.2.12)' (Gem::UnsatisfiableDepedencyError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/dependency_resolver.rb:286:in `resolve'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/request_set.rb:39:in `resolve'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/request_set.rb:47:in `resolve_current'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems.rb:244:in `detect_gemdeps'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems.rb:1109:in `<top (required)>'
	from <internal:gem_prelude>:1:in `require'
	from <internal:gem_prelude>:1:in `<compiled>'
$ gem --version
Invalid gemspec in [rails_config.gemspec]: uninitialized constant Gem::Specification::Date
Unable to resolve dependency: user requested 'shoulda (>= 0)'
You may need to `gem install -g` to install missing gems

2.4.6
~~~

**Comment by rewritten** [comment]
any workaround while this is solved?

**Comment by indirect** [comment]
For the time being, we recommend working around this by using Bundler and `bundle exec` or binstubs instead of setting RUBYGEMS_GEMDEPS.

**Comment by franklinyu** [comment]
So my workaround for Homebrewer:

```shell
alias brew="RUBYGEMS_GEMDEPS='' brew"
```

😂

**Comment by drbrain** [comment]
Note that `export RUBYGEMS_GEMDEPS=-` is the same as putting `bundle exec` in front of every ruby executable.

**Comment by bronzdoc** [comment]
I can't reproduce this with rubygems 2.6.7.
Testing with the Gemfile provided and vagrant, setting  RUBYGEMS_GEMDEPS='-' ,  is this issue still valid?

**Comment by franklinyu** [comment]
I think this issue is solved in 2.6.7. I just updated from 2.6.6 to 2.6.7. I only tried Homebrew though.

**Comment by bronzdoc** [comment]
I'm going to close this, please reopen if this issue persist in rubygems >= 2.6.7
